setup.py

```diff
@@ -171,6 +171,17 @@
     print("32-bit Windows Python runtime is not supported. Please switch to 64-bit Python.")
     sys.exit(-1)
 
+import platform
+python_min_version = (3, 6, 1)
+python_min_version_str = '.'.join((str(num) for num in python_min_version))
+python_max_version = (3, 9, 0)
+python_max_version_str = '.'.join((str(num) for num in python_max_version))
+if sys.version_info < python_min_version or sys.version_info >= python_max_version:
+    print("You are using Python {}. Python >={},<{} is required.".format(platform.python_version(),
+                                                                         python_min_version_str,
+                                                                         python_max_version_str))
+    sys.exit(-1)
```

Comment:
Hmm, why not use some python3 features like f-strings here?
```suggestion
def tuple2str(t, sep='.'):
   return sep.join(map(str, t))
python_min_version = (3, 6, 1)
python_max_version = (3, 9, 0)
if sys.version_info < python_min_version or sys.version_info >= python_max_version:
    print(f"You are using Python {tuple2str(sys.version_info[:3]}. Python >={tuple2str(python_min_version)},<{tuple2str(python_max_version)} is required.")
    sys.exit(-1)
```

Reply:
You would receive syntax error if Python is < 3.6, and that's exactly the situation where we are trying to have a reasonable error message.